src/Simulator/SandSimulator.cpp

```diff
@@ -62,7 +62,7 @@ void cSandSimulator::SimulateChunk(std::chrono::milliseconds a_Dt, int a_ChunkX,
 			);
 			*/
 
-			auto FallingBlock = cpp14::make_unique<cFallingBlock>(Pos, BlockType, a_Chunk->GetMeta(itr->x, itr->y, itr->z));
+			auto FallingBlock = cpp14::make_unique<cFallingBlock>(Vector3d(0.5, 0, 0.5) + Pos, BlockType, a_Chunk->GetMeta(itr->x, itr->y, itr->z));
```

Comment:
Mightn't it be slightly better to let cFallingBlock handle the specifics of adding 0.5 here? Also now World.h sees some duplication when it does the same offset.

Reply:
That's true, thanks for the idea. I really shouldn't code so late at night :)

Reply:
Wait, no, it won't work, I'd have to duplicate the body of `cWorld:SpawnFallingBlock` in order to do that properly. I think I'll keep it as-is.

Reply:
Template the first parameter of SpawnFallingBlock, you know you want to...

Or at least get this line to call m_World->SpawnFallingBlock, that should work right

Reply:
This should just call into m_World->SpawnFallingBlock

Reply:
Tested, replacement works.

Reply:
Yes, the original code didn't remove the sand block if creating the entity failed, but I decided against that behavior.